gnomad_constraint/utils/constraint_basics.py

```diff
@@ -165,20 +167,221 @@ def get_proportion_observed_by_coverage(
+    :param pops: List of populations. Defaults to `POPS`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants'.
+    :return: A Dictionary of intercepts and slopes for the full plateau models.
+    """
+    pop_lengths = get_all_pop_lengths(ht, pops)
+
+    plateau_models = build_plateau_models_total(ht, weighted=weighted)
+
+    if pop_anal:
+        arg_expr = {}
+        for length, pop in pop_lengths:
+            plateau_model = build_plateau_models_pop(ht, pop, length, weighted=weighted)
+            arg_expr[pop] = plateau_model[pop]
+        plateau_models = plateau_models.annotate(**arg_expr)
+
+    return plateau_models
+
+
+def build_plateau_models_pop(
+    ht: hl.Table, pop, length, weighted: bool = False
+) -> Dict[str, Tuple[float, float]]:
+    """
+    Calibrate high coverage model for each population in `POPS`.
+
+    :param ht: High coverage Table.
+    :param pop: List of populations. Defaults to `POPS`.
+    :param length: The indices of each population.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for plateau models of each population.
+    """
+    agg_expr = {
+        pop: [
+            hl.agg.group_by(
+                ht.cpg,
+                hl.agg.linreg(
+                    ht[f"observed_{pop}"][i] / ht.possible_variants,
+                    [1, ht.mu_snp],
+                    weight=ht.possible_variants if weighted else None,
+                ).beta,
+            )
+            for i in range(length)
+        ]
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_plateau_models_total(
+    ht: hl.Table, weighted: bool = False
+) -> Dict[str, Tuple[float, float]]:
+    """
+    Calibrate high coverage model in total.
+
+    :param ht: High coverage Table.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for a plateau model.
+    """
+    agg_expr = {
+        "total": hl.agg.group_by(
+            ht.cpg,
+            hl.agg.linreg(
+                ht.observed_variants / ht.possible_variants,
+                [1, ht.mu_snp],
+                weight=ht.possible_variants if weighted else None,
+            ).beta,
+        )
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_coverage_model(coverage_ht: hl.Table) -> (float, float):
+    """
+    Calibrate coverage model.
+
+    This function uses linear regression to build a model of log10(coverage) to this scaled ratio as a correction
+    factor for low coverage sites.
+
+    .. note::
+        The following annotations should be present in `coverage_ht`:
+            - low_coverage_obs_exp - an observed:expected ratio for a given coverage level
+            - log_coverage - log10 coverage
```

Comment:
to make this function more general, it might be nice to pass these annotations in as arguments, e.g.:
```
return tuple(
        coverage_ht.aggregate(
            hl.agg.linreg(
                coverage_ht[low_cov_oe_field], [1, coverage_ht[log_cov_field]]
            )
        ).beta
    )

```
also slight preference to call the input Table ht and just note in the docstring that it's a coverage HT (like you did above in the build plateau models functions)